azurerm/internal/services/privatedns/tests/resource_arm_private_dns_txt_record_test.go

```diff
@@ -0,0 +1,334 @@
+package tests
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/services/privatedns/mgmt/2018-09-01/privatedns"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/acceptance"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/features"
+)
+
+func TestAccAzureRMPrivateDnsTxtRecord_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_private_dns_txt_record", "test")
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMPrivateDnsTxtRecordDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMPrivateDnsTxtRecord_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMPrivateDnsTxtRecordExists(data.ResourceName),
+					resource.TestCheckResourceAttrSet(data.ResourceName, "fqdn"),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func TestAccAzureRMPrivateDnsTxtRecord_requiresImport(t *testing.T) {
+	if !features.ShouldResourcesBeImported() {
+		t.Skip("Skipping since resources aren't required to be imported")
+		return
+	}
+
+	data := acceptance.BuildTestData(t, "azurerm_private_dns_txt_record", "test")
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMPrivateDnsTxtRecordDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMPrivateDnsTxtRecord_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMPrivateDnsTxtRecordExists(data.ResourceName),
+				),
+			},
+			data.RequiresImportErrorStep(testAccAzureRMPrivateDnsTxtRecord_requiresImport),
+		},
+	})
+}
+
+func TestAccAzureRMPrivateDnsTxtRecord_updateRecords(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_private_dns_txt_record", "test")
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMPrivateDnsTxtRecordDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMPrivateDnsTxtRecord_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMPrivateDnsTxtRecordExists(data.ResourceName),
+					resource.TestCheckResourceAttr(data.ResourceName, "record.#", "2"),
+				),
+			},
+			{
```

Comment:
Let's add import steps between all these checks and one more step at the end that takes us back to the basic configuration just to confirm it all works

Reply:
Great, thanks! Hoping I've interpreted your suggestion correctly, have pushed some changes.